### ISSUE TITLE
UX improvements for the expand mechanism

### ISSUE THREAD

**Issue by piegamesde** [NixOS/nixos-search#277]
UX of the expand mechanism can be improved in many points:

- [ ] It's not navigable by keyboard
- [ ] The button is a lot smaller and thus harder to hit than before (Fitts' law)
- [ ] The button says "Show/Hide package details" even when searching options
- [ ] Placing the button onto the separator line is a weird choice. This makes it feel like the button is not related to one search entry, but *between* two. It also isn't 100% clear whether it refers to the item above or below the line.

------

From <https://github.com/NixOS/nixos-search/issues/272#issuecomment-766807288_>. Sorry @garbas, no medal for me today ^^

**Comment by garbas** [comment]
@piegamesde Do you maybe have some suggestion how would you fix the expand mechanism?

**Comment by piegamesde** [comment]
Did the old site do something wrong in that regard? If I remember correctly, the whole list item was a button and it worked just fine. Alternatively, I see two approaches:

1. Related to the previous way of doing it, have a toggle button on the side that expands the item down (think of Accordion menu)
2. Have a "show more"/"show less", which would be more similar to the current implementation. Important there:
    - Make the button permanently visible and accessible by keyboard
    - Move the button *above* the separator line

**Comment by turboMaCk** [comment]
Sorry for jumping in but I think I might be best equipt to clarify few things.

Before I do so I want to make it clear that we're glad that you've opened the issue because every data-point counts. That's being said for sake of tracking all the information I'll link few things related to the points.

> It's not navigable by keyboard

This is not true. In fact the reason why the whole item is no longer clickable is to make keyboard navigation possible. See this issue: https://github.com/NixOS/nixos-search/pull/259 . When whole item was clickable browser was unable to find any focusable element so keyboard navigation was not working. Fix to this issue than caused this issue https://github.com/NixOS/nixos-search/issues/271 since having the focusable button element also means it's not possible to select text within the element.

TL;DR issue like this are hard to resolve because different people care about different things and it's hard to keep everyone happy.

Anyway the keyboard selection does in fact work. as you can see:

![keyboard-selection](https://user-images.githubusercontent.com/2130305/105764054-c3b24b00-5f56-11eb-84c0-e3547feb55f3.gif)

All you need to do is to use tab to select the title of item you want to expand and then press enter.

> The button is a lot smaller and thus harder to hit than before (Fitts' law)

you can still click on title as far as I know before there was no button at all and thing opened by clicking on item. But then it was hard to select text - see https://github.com/NixOS/nixos-search/pull/261#issuecomment-758348509

> Placing the button onto the separator line is a weird choice

This was raised before as well in https://github.com/NixOS/nixos-search/pull/261#pullrequestreview-573543088 and we hope to address it soon

**Comment by garbas** [comment]
Doing some issue maintenance while I plan my work. The expand mechanism is going to be address with #286 .
